### DISCUSSION TITLE
Arbitrary `group-*` selector?

### DISCUSSION THREAD

**Discussion by AssisrMatheus** [tailwindlabs/tailwindcss#9378] [Help] [answered]
We usually can use the `group` selector to propagate the state to inner elements, example:

```jsx
<button className='group flex items-center justify-center'>
    <span className='group-hover:text-green-500'>Example</span>
    <Icon className='group-hover:text-pink-500' />
</button>
```

But what if, in this case, a button could be `selected` and now I want to style my children elements based on this `selected` status, and give `group-[selected]` to the parent. Something in the likes of:

```jsx
<button className={`group flex items-center justify-center ${isSelected ? 'group-[selected]' : ''}`}>
    <span className='group-hover:text-green-500 group-[selected]:text-green-500'>Example</span>
    <Icon className='group-hover:text-pink-500 group-[selected]:text-pink-500' />
</button>
```

Is that achievable?

**Comment by wongjn** [comment, marked as answer]
You could use an arbitrary variant:
```jsx
<button className={`group flex items-center justify-center ${isSelected ? 'selected' : ''}`}>
    <span className='group-hover:text-green-500 [.selected>&]:text-green-500'>Example</span>
    <Icon className='group-hover:text-pink-500 [.selected>&]:text-pink-500' />
</button>
```

**Comment by AssisrMatheus** [comment]
Thank you! Is that a new feature? I'm not managing to have it working, unfortunately:
![CleanShot 2022-09-20 at 10 57 05](https://user-images.githubusercontent.com/12276174/191292557-9e16e3b3-496b-4845-8874-2d7b0e68d4fc.png)
![CleanShot 2022-09-20 at 10 57 54](https://user-images.githubusercontent.com/12276174/191292634-6a7419c9-8a2d-4041-8874-ab6a0bd09117.png)

The `1/2` was supposed to be pink. Tailwind version is `3.0.23`

I managed to have it working using the playground so i'm probably missing something

-----

Edit: Updated to newest tailwind and it worked. I was having issues when updating but after fixing those, everything works fine. Thank you!
